plugin/vcs/github/github.go

```diff
@@ -344,9 +354,85 @@ func (p *Provider) ExchangeOAuthToken(ctx context.Context, _ string, oauthExchan
 	return oauthResp.toVCSOAuthToken(), nil
 }
 
-// FetchAllRepositoryList fetches all repositories where the authenticated user has a maintainer role.
-func (p *Provider) FetchAllRepositoryList(ctx context.Context, oauthCtx common.OauthContext, instanceURL string) ([]*vcs.Repository, error) {
-	return nil, errors.New("not implemented yet")
+// FetchAllRepositoryList fetches all repositories where the authenticated user
+// has a owner role, which is required to create webhook in the repository.
+//
+// Docs: https://docs.github.com/en/rest/repos/repos#list-repositories-for-the-authenticated-user
+func (p *Provider) FetchAllRepositoryList(ctx context.Context, oauthCtx common.OauthContext, _ string) ([]*vcs.Repository, error) {
+	var githubRepos []Repository
+	page := 1
+	for {
+		repos, hasNextPage, err := p.fetchPaginatedRepositoryList(ctx, oauthCtx, page)
+		if err != nil {
+			return nil, errors.Wrap(err, "fetch paginated list")
+		}
+		githubRepos = append(githubRepos, repos...)
+
+		if !hasNextPage {
+			break
+		}
+		page++
+	}
+
+	var allRepos []*vcs.Repository
+	for _, r := range githubRepos {
+		allRepos = append(allRepos,
+			&vcs.Repository{
+				ID:       r.ID,
+				Name:     r.Name,
+				FullPath: r.FullName,
+				WebURL:   r.HTMLURL,
+			},
+		)
+	}
+	return allRepos, nil
+}
+
+// fetchPaginatedRepositoryList fetches repositories where the authenticated
+// user has a owner role in given page. It return the paginated results along
+// with a boolean indicating whether the next page exists.
+func (p *Provider) fetchPaginatedRepositoryList(ctx context.Context, oauthCtx common.OauthContext, page int) (repos []Repository, hasNextPage bool, err error) {
+	// We will use user's token to create webhook in the project, which requires the
+	// token owner to be at least the project maintainer(40).
+	url := fmt.Sprintf("%s/user/repos?affiliation=owner&page=%d&per_page=100", apiURL, page)
```

Comment:
nit: extract 100 to be a constant.

Reply:
Addressed in https://github.com/bytebase/bytebase/pull/1827/commits/98218dd4c30d6eda93443084d26b4c0f62c67bfd